### PR TITLE
Implement REST API to get deployment by environment ident

### PR DESCRIPTION
**Pull request by maheshkelkar**
References #169 
References #21

### CODE REVIEW

**Comment by maheshkelkar** [comment]
@GiriDandu please review when Travis is green

File: src/main/groovy/deploydb/resources/EnvironmentResource.groovy [hunk 1]
```diff
@@ -54,12 +57,36 @@ public class EnvironmentResource {
...
+    @UnitOfWork
+    @Timed(name = "get-requests")
+    List<Deployment> getDeployments(@PathParam("name") String environmentIdent,
+                                    @QueryParam("pageNumber") @DefaultValue("0")
+                                            IntParam pageNumber,
+                                    @QueryParam("perPageSize") @DefaultValue("20")
+                                            deploydb.ModelPageSizeParam perPageSize) {
+        List<Deployment> deploymentList = this.workFlow.deploymentDAO
+                .getByEnvironmentIdent(environmentIdent, pageNumber.get(), perPageSize.get())
+        if (deploymentList == null || deploymentList.isEmpty()) {
```
Comment (GiriDandu) [inline src/main/groovy/deploydb/resources/EnvironmentResource.groovy]: In groovy you can do this if (deploymentList) that will check both for null and empty

File: src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy [hunk 2]
```diff
@@ -31,4 +45,46 @@ class DeploymentDAOSpec extends Specification {
...
+        integModelHelper.sendCreateArtifact()
+
+        when:
+        List<Deployment> deploymentsByEnv
+        List<Deployment> allDeployments
+        List<Deployment> deploymentsByUnknownEnv
+        integAppHelper.withSession {
+            deploymentsByEnv = integAppHelper.runner.getApplication().workFlow.deploymentDAO
+                    .getByEnvironmentIdent("basicEnv", 0, 20)
+            allDeployments = integAppHelper.runner.getApplication().workFlow.deploymentDAO
```
Comment (GiriDandu) [inline src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy]: Why do we have this "allDeployments" in this function?

File: src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy [hunk 3]
```diff
@@ -31,4 +45,46 @@ class DeploymentDAOSpec extends Specification {
...
+                    .getByPage(0, 20)
+            deploymentsByUnknownEnv = integAppHelper.runner.getApplication().workFlow.deploymentDAO
+                    .getByEnvironmentIdent("integ", 0, 20)
+        }
+
+        then:
+        deploymentsByEnv.size() == 1
```
Comment (GiriDandu) [inline src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy]: Can we split this into three different tests

File: src/main/groovy/deploydb/dao/DeploymentDAO.groovy [hunk 4]
```diff
@@ -58,4 +58,21 @@ class DeploymentDAO extends AbstractDAO<Deployment> {
                                          .list()
         return deployments
     }
+
+    /**
+     * Find deployments for an environment
+     *
+     * @param environmentIdent ident of the Environment
```
Comment (rtyler) [inline src/main/groovy/deploydb/dao/DeploymentDAO.groovy]: The name of the parameter here, and the name of the parameter in the actual method declaration aren't the same. D'oh!

File: src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy (hunk 2 above)
Comment (maheshkelkar) [inline src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy]: I have changed this. It simply makes sure that contents of getByPage & getByEnvironmentIdent are same. Plus its not null. Hence its part of same test

File: src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy (hunk 3 above)
Comment (maheshkelkar) [inline src/test/groovy/deploydb/dao/DeploymentDAOSpec.groovy]: I have removed deploymentsByUnknownEnv here

File: src/main/groovy/deploydb/dao/DeploymentDAO.groovy (hunk 4 above)
Comment (maheshkelkar) [inline src/main/groovy/deploydb/dao/DeploymentDAO.groovy]: done

File: src/main/groovy/deploydb/resources/EnvironmentResource.groovy (hunk 1 above)
Comment (maheshkelkar) [inline src/main/groovy/deploydb/resources/EnvironmentResource.groovy]: done

**Comment by maheshkelkar** [comment]
@rtyler @GiriDandu I will merge it in half an hour if I have your blessings

**Comment by GiriDandu** [comment]
:+1: from my side

**Pull request merged**